### DISCUSSION TITLE
Source code blocks list of supported languages

### DISCUSSION THREAD

**Discussion by refaelsh** [nvim-orgmode/orgmode#380] [Q&A] [answered]
Hi,

I have a question please: where is the lost of Source code blocks supported languages?
This is an example for the Nix language.
```
#+BEGIN_SRC nix
```

What are some other languages?

**Comment by jgollenz** [comment, marked as answer]
Try `:TSInstallInfo` to list all available ~~languages~~ grammars of treesitter. To install, `:TSInstall nix`

**Reply by refaelsh** [reply]
Thank you :-)

**Comment by refaelsh** [comment]
A follow-up question, please. As I understand, when I move to a new machine, and take my `init.vim` with me of course, I need to remember to manually run `:TSInstall nix`, right? If yes, then this is very bad, I so much like automation and I will definitely forget to run this and will bang my had why nothing works.

**Reply by jgollenz** [reply]
Yes, but there for sure is a way to automate this. If not in nvim itself, then via a shell-script where you call `nvim -c "TSInstall <a list of all your grammars>"`. I'd need that myself at one point, but unfortunately I have no time to look into it now. Please share if you find a nice approach :+1:

**Reply by refaelsh** [reply]
I am planning to move my init.vim to be 100% in lua, and then I will try to write something in lua that will make this automatic.

P.S.
> shell-script where you call nvim -c "TSInstall <a list of all your grammars>"

Then I will need to remember to manually run some shell script :-)

**Reply by jgollenz** [reply]
I'm not sure if that's something you want in your `init.lua`. Maybe if you have a single grammar (nix in your case) that you want to install it's fine, but I have like twenty. No point in trying to install (or even checking if they are installed) on every open of neovim. Since you were talking about a new machine, can't you just automatically call a script that will install all the grammars when the machine is set up? As far as I know nix is quite handy for that, right? But ofc it's annoying if you have different distros on your machines. But if all fails, you can always make a cronjob, no?

**Comment by refaelsh** [comment]
> As far as I know nix is quite handy for that, right?

Yes, but, still, this is gonna a one time, manual intervention that I will definitely forget.

> you can always make a cronjob

Hmm... Let me think about it.

**Reply by jgollenz** [reply]
What do you mean by manual intervention? You have to add that to your setup routine once and then every time you setup a new machine it will be executed, or am I missing something?

**Reply by refaelsh** [reply]
> every time you setup a new

This part, I dont know how to do in NixOS, maybe its even impossible.

**Reply by jgollenz** [reply]
I see. I'm not a Nix user myself, but I was under the impression that it is capable of doing such a thing. Please,let me know if I'm wrong ;)

**Reply by Maltimore** [reply]
Maybe I'm missing something, but you can just indicate in your vimrc which TreeSitter languages to install, why don't you do that? Are neither of you using the nvim-treesitter plugin (which will be migrated to neovim core at some point)?
I have in my vimrc

```lua
require'nvim-treesitter.configs'.setup {
  ensure_installed = {"c", "cpp", "python", "org", "latex", "lua", "vim", "html", "css", "yaml", "toml", "json"},
  highlight = {
    enable = true,
  },
  indent = {
    enable = true,
  },
}
```

**Reply by refaelsh** [reply]
Brilliant. That's exactly what I need. Thank you very much! :-)

**Reply by refaelsh** [reply]
On second thought, it's almost perfect. What if I want all languages installed?

**Reply by Maltimore** [reply]
There's a way to do it: just use "all"

**Reply by refaelsh** [reply]
Can I hug you, please? :-)

Thank you very much.

**Reply by jgollenz** [reply]
@Maltimore thanks a lot :) I am using the treesitter plugin, but I haven't found time to properly go through the config, just did what was necessary to get orgmode running :grin:

**Reply by teto** [reply]
nix definitely does this.

**Reply by refaelsh** [reply]
> nix definitely does this.

Please elaborate.
